core/sup_tree_core/executor_pool/async_job_runner.ex

```diff
@@ -142,7 +142,7 @@ defmodule AntikytheraCore.ExecutorPool.AsyncJobRunner do
 
   defp job_succeeded(%{queue_name: queue_name, job_key: job_key} = state) do
     report_on_finish(state, Time.now(), "success")
-    Queue.remove_locked_job(queue_name, job_key)
+    if queue_name, do: Queue.remove_locked_job(queue_name, job_key)
```

Comment:
It's very important that now `queue_name` can be `nil` depending on the execution mode.
Let's add comment on l.57 (when filling out `:queue_name` field) as follows:
```ex
queue_name: queue_name, # can be `nil` if the job is executed in `:immediate` mode
```
(Note that, as currently discussed, the name of the option can be changed.)

Reply:
In addition to that, we should change typespec of `run/4`:
`queue_name :: v[atom]` => `queue_name :: v[nil | atom]`
(`nil` is an `atom` in a type sense but we should distinguish `nil` from other atoms)

Reply:
Fixed in 0a17fb1cc0e34322e8281898a5ec061cef1ecb01